src/_cffi_src/openssl/dsa.py

```diff
@@ -38,9 +38,81 @@
 """
 
 MACROS = """
+/* DSAparams_dup is a macro in 0.9.8 */
+DSA *DSAparams_dup(DSA *);
 int DSA_generate_parameters_ex(DSA *, int, unsigned char *, int,
                                int *, unsigned long *, BN_GENCB *);
+/* added in 1.1.0 to access the opaque struct */
+void DSA_get0_pqg(const DSA *, BIGNUM **, BIGNUM **, BIGNUM **);
+int DSA_set0_pqg(DSA *, BIGNUM *, BIGNUM *, BIGNUM *);
+void DSA_get0_key(const DSA *, BIGNUM **, BIGNUM **);
+int DSA_set0_key(DSA *, BIGNUM *, BIGNUM *);
```

Comment:
Ah, good point. Since they're functions (either available in 1.1.0 or declared by us) we can put them in FUNCTIONS unconditionally. This same issue is present in the RSA PR.